pkg/registry/core/pod/strategy_test.go

```diff
@@ -371,7 +371,10 @@ func TestWaitingForGatesCondition(t *testing.T) {
 				}
 			}
 
-			if diff := cmp.Diff(tt.want, got); diff != "" {
+			if got.LastTransitionTime.IsZero() && got.Type != "" {
```

Comment:
more like this:
```
if got.Type == "" {
  t.Errorf("Scheduled condition not found")
} else {
  check lastTransitionTimestamp
  check diff
}
```

Reply:
In the former test cases `pod without .spec.schedulingGates` or `pod with feature disabled`, the `got.Type` will always be empty string as expected, no need to error out.
Only when got.Type is not empty, which indicates pod with .spec.schedulingGates and gate feature enabled, we need to check if lastTransitionTimestamp is null.

Reply:
ah, gotcha

/approve
/lgtm
/hold cancel

Reply:
#122651 done

Reply:
@alculquicondor I guess the /approve label was not added unless you send it directly on comment instead of review comment.

Reply:
Strange, why this needs sig node approvers, I thought it's sig-apis.